src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -104,6 +104,11 @@ private function addService($id, $definition)
             $code .= sprintf("        synchronized: true\n");
         }
 
+        if ($definition->isDeprecated()) {
+            $code .= "        deprecated: true\n";
+            $code .= sprintf("        deprecation_message: %s\n", $definition->getDeprecationMessage('%service_id%'));
```

Comment:
I propose naming this one: `deprecation_template`. This maps to the fact that we enforce a `%service_id%` placeholder.